### PR TITLE
Mod categories

### PR DESCRIPTION
**Pull request by OzoneH3**
Inventory like categories.

Are the categories ok like that or should I add some more/move some mods around?

```
ITEM ADDITION MODS
    Craftable Gun Pack
    DeadLeaves' Fictional Guns
    Generic Guns
    Icecoon's Weapons Pack
    Medieval and Historic Content
    More Survival Tools
    Mythological Replicas
CREATURE MODS
    DinoMod
BUILDINGS MODS
    Animatronic Monsters
    Tall Buildings
VEHICLE MODS
    Boats
    Folding Parts pack
    Tanks and Other Vehicles
    Vehicle Additions Pack
REBALANCING MODS
    Classic Roguelike Classes
    Fast Zombies
    Prevent Zombie Revivication
    Slow Zombies
    StatsThoughSkills
    Tough Zombies
    Zombie Nightvision
MAGICAL MODS
    Arcana and Magic Items
    Necromancy
EXCLUSION MODS
    Disable mutagens
    Disable religious texts
    Mundane Zombies
    No Antique Firearms
    No Energy Weapons
    No Fictional Guns
    No Flaming Weapons
    No Fungal Monsters
    No Joke Monsters
    No Medieval Items
    No Monsters
    No Powered Armor
    No Survivor Armor
```

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
This is glorious.

**Comment by chaosvolt** [comment]
Ooh, nice. o.o

**Comment by kevingranade** [comment]
Awesome!

This isn't a feature request for this PR (but if you feel like it, that'd
be great too). How hard do you think it'd be to do filtering based on these
categories? I'd like to have some filtering, probably tab-like, but I'm not
picky about the details, to keep the blacklist mods from flooding the mod
selection display.

File: data/mods/Animatronics/modinfo.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
         "ident": "Animatronics",
         "name": "Animatronic Monsters",
         "description": "Adds a chain of bankrupt pizzerias, with still functioning animatronics inside.",
+	"category": "buildings",
```
Comment (Coolthulhu) [inline data/mods/Animatronics/modinfo.json]: Tabs

**Comment by OzoneH3** [comment]
@kevingranade You mean something like the options tabs where you can designate which category schould be in which tab? That would be pretty easy.

@Rivet-the-Zombie No jumping around right now :(

The only real problem I have with both requests is that there aren't that much (default moving through the menu) keys left.

Tab, < and > move through the top most tabs. Left and Right arrow switch between the mod list.
What would be the keys to change the mod list tabs/jump between categories?

File: src/mod_manager.cpp [hunk 2]
```diff
@@ -18,6 +18,16 @@
 #define MOD_SEARCH_FILE "modinfo.json"
 
 static std::unordered_set<std::string> obsolete_mod_list;
+static const std::vector<std::pair<std::string, std::string> > categories = {
+    {"weapons", _("WEAPON MODS")},
```
Comment (BevapDin) [inline src/mod_manager.cpp]: I suspect this wont translate correctly. The static vector is initialized before `main` is called and therefor before the translation system is set up. You can circumvent this by moving it into a function:
```C++
const vector &get_categories() {
    static const vector categories = ...;
    return categories;
}
```

The vector in the function is initialized when the function is called the first time, which should be after the translation system has been initialized.

Comment (OzoneH3) [inline src/mod_manager.cpp]: Good point, changed that in the other PR.

**Comment by BevapDin** [comment]
The center column (the one with the blue and white "-" and "+" characters, only visible when the mod list on the right is active) does not not line up anymore. I think it is supposed to show how/whether a mod can be moved in the list: a white "-" / "+" means the mod can not be moved that way in the list, a blue means it can. (Try this also with the "Tanks and Other Vehicles" mod, it depends on the "Vehicle Additions Pack" and can therefor not be moved before that mod.)

**Comment by Coolthulhu** [comment]
The functionality itself looks good, but the categories could use some extra work:
* Animatronics fits creatures more than buildings (even though it fits buildings too)
* No zombie revival needs to be in the same category as zombie mutators, like slow zeds. It doesn't blacklist anything.
* More Survival Tools has some weapons, but doesn't fit the weapon category
* There are only two mods in "character" category, meaning it doesn't help with categorization
* There are only two mods in "magical" category

I'd generalize the "weapons" to "item additions" and create a separate category for mods that don't add or remove anything but alter existing behavior ("rebalancing" or "conversions") - fast/slow zeds, no zed rez, statsthroughskills, foldable parts, generic guns etc.

**Comment by OzoneH3** [comment]
Updated the fist post with the current categories and their items.

**Pull request merged**